File: echo_tool.py
from pathlib import Path
from check_file_permission import check_file_permission

def echo_text(text, file_path, config, append=False):
    if file_path:
        if not check_file_permission(file_path, "write", config):
            raise PermissionError(f"Write access denied for {file_path}")
        file_path = Path(file_path).resolve()
        mode = "a" if append else "w"
        with open(file_path, mode, encoding="utf-8") as f:
            f.write(text + "\n")
        return f"Wrote to {file_path}"
    return text

File: ts-tools/echo_tool.ts
/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function echoText(
  text: string,
  filePath?: string,
  config?: any,
  append: boolean = false,
): Promise<string> {
  return new Promise((resolve, reject) => {
    if (filePath && config) {
      checkFilePermission(filePath, 'write', config)
        .then(() => {
          const resolvedPath = path.resolve(filePath);
          const mode = append ? 'a' : 'w';
          fs.writeFile(resolvedPath, text + '\n', { mode }, (err) => {
            if (err) {
              reject(err);
            } else {
              resolve(`Wrote to ${filePath}`);
            }
          });
        })
        .catch(reject);
    } else {
      resolve(text);
    }
  });
}
